### PR TITLE
docs: remove demo-app docs

### PR DESCRIPTION
**Pull request by ajcasagrande**
**If your build fails**  due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-go/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [X] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [X] I am not introducing a new dependency (add notes below if you are)
- [N/A] I have added unit tests for the new feature or bug fix (if not, why?)
- [N/A] I have fully tested (add details below) this the new feature or bug fix (if not, why?)
- [N/A] I have opened a PR for the related docs change (if not, why?)
  <link to docs PR>

### CODE REVIEW

**Review by presatish — APPROVED** [review]
README.md file referred in some places and the links to it are broken for some reason. For e.g. https://github.com/EdgeX-Camera-Management/device-onvif-camera/blob/remove-demo-app-docs/doc/general-usage.md#next-steps

**Review by presatish — CHANGES_REQUESTED** [review]
Need some minor changes.

File: README.md [hunk 1]
```diff
@@ -31,8 +31,6 @@ The figure below illustrates the software flow through the architecture componen
 
 # Getting Started
 
-Learn how to configure and run the service by following these [instructions](./doc/setup.md).
```
Comment (ChristianDarr-personal) [inline README.md:34]: Same comment as other people, what do we want to do with the files left dangling from removing this line?

Comment (presatish) [reply README.md:34]: I think for now we can leave these files as it is at least until we migrate everything to edgex docs.

**Review by presatish — APPROVED** [review]

**Pull request merged**